src/settings/BatchGroupConfigurationSettings/BatchGroupConfigurationSettings.js

```diff
@@ -31,14 +43,87 @@ const BatchGroupConfigurationSettings = ({ mutator }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   []);
 
-  const onSaveExportConfig = useCallback(
-    () => {
+  const fetchExportConfig = useCallback(
+    (id) => {
+      setIsLoading(true);
+      setExportConfig();
+      setCredentials();
 
+      mutator.exportConfig.GET({
+        params: {
+          query: `batchGroupId==${id}`,
+        },
+      })
+        .then(exportConfigResp => {
+          if (exportConfigResp.length) {
+            const exportConfigId = exportConfigResp[0].id;
+
+            setExportConfig(exportConfigResp[0]);
+            mutator.exportConfigId.update({ id: exportConfigId });
+
+            return mutator.credentials.GET();
+          } else {
+            setExportConfig({});
+
+            return Promise.resolve();
+          }
+        })
+        .then(credentialsResp => {
+          if (credentialsResp) setCredentials(credentialsResp);
+          else setCredentials({});
```

Comment:
I don't think it's a good idea to use such extravagant condition's code form.

Reply:
I would make structure of this promise chain more clear and simple - 
```
fetchConfigs()
  .then(setConfigsAndFetchCreds)
  .then(setCreds)
  .finally;
```

Reply:
Made it more clear and simple, thx